### PR TITLE
test: add notebook with remote function use cases

### PR DESCRIPTION
**Pull request by shobsi**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: notebooks/remote_functions/remote_function_usecases.ipynb [hunk 1]
```diff
@@ -0,0 +1,1134 @@
+{
```
Comment (tswast) [inline notebooks/remote_functions/remote_function_usecases.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    @bpd.remote_function()
</pre><p>Let's add a comment here explaining this is for demonstration purposes only and for this use case one should prefer using case_when. (<a href="https://cloud.google.com/python/docs/reference/bigframes/latest/bigframes.series.Series#bigframes_series_Series_case_when" target="_blank">https://cloud.google.com/python/docs/reference/bigframes/latest/bigframes.series.Series#bigframes_series_Series_case_when</a>)</p><p> </p><p>For your reference:</p>

<pre class="ql-syntax" spellcheck="false">duration_cat = df['duration_minutes'].case_when(
     caselist=[
         (df['duration_minutes'].lt(90), "short"),
         (df['duration_minutes'].lt(180), "medium"),
         (True, "long"),
    ]
)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/googleapis/python-bigquery-dataframes/pull/808/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (tswast) [inline notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>I imagine we should have some explanation about the condition and/or what happens with these references. For example, they won't change after the function is deployed.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/googleapis/python-bigquery-dataframes/pull/808/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (tswast) [inline notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>Likewise, some explanation that this is only for built-in and there is an example of third-party packages later on.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/googleapis/python-bigquery-dataframes/pull/808/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (tswast) [inline notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>Any restrictions to be aware of here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/googleapis/python-bigquery-dataframes/pull/808/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (tswast) [inline notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>Explain this is in <code>pip</code> format and can include the version number. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/googleapis/python-bigquery-dataframes/pull/808/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (shobsi) [reply notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>Ran into couple of issues (b/349926559, b/349927177), but noted as a broad comment that the use cases are only for demo purpose of remote_function and not necessarily the best way to achieve the end result.</p>

Comment (shobsi) [reply notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>Added.</p>

Comment (shobsi) [reply notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>Added.</p>

Comment (shobsi) [reply notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>Nothing that I can think of. I made this use case a bit more generic by chaining function dependencies outside the function body, PTAL.</p>

Comment (shobsi) [reply notebooks/remote_functions/remote_function_usecases.ipynb:1]: <p>Done.</p>

**Review by tswast — APPROVED** [review]

**Pull request merged**